src/pkg/reg/adapter/harbor/base/adapter.go

```diff
@@ -182,9 +182,14 @@ func (a *Adapter) PrepareForPush(resources []*model.Resource) error {
 		return errors.Wrapf(err, "list projects with query %s", q)
 	}
 
+	proxyCacheProjects := make(map[string]*Project)
 	existProjects := make(map[string]*Project)
 	for _, p := range queryProjects {
 		existProjects[p.Name] = p
+		// if project with registry_id, that means this is a proxy cache project.
+		if p.RegistryID > 0 {
```

Comment:
we don't have any other attribution to judge this but the registry id?

Reply:
Discuss with @stonezdj @AllForNothing, no other attribution can identify proxy cache project, same judgement with UI.